### PR TITLE
Add support for FreeSurfer license.

### PR DESCRIPTION
**Pull request by natacha-beck**

### CODE REVIEW

File: cbrain_task/recon_all/bourreau/recon_all.rb [hunk 1]
```diff
@@ -139,7 +147,23 @@ def cluster_commands #:nodoc:
 
     recon_all_command = "recon-all#{lbl_ext} #{with_qcache} #{with_mprage} #{with_3T_data} #{with_cw256} #{with_hippocampal} -sd . #{subjid_info} #{step} #{with_notal_check} #{lbl_options}"
 
+    # Copy license
+    cp_license = <<-CP_LICENSE
+
+# Handle FreeSurfer license
+
+if [ ! -f "$FREESURFER_HOME/license.txt" ] && [ ! -f "$FREESURFER_HOME/.license" ] ; then
+  echo Could not find a license installed.
```
[the remark this answers is not in the document]
Comment (natacha-beck) [reply cbrain_task/recon_all/bourreau/recon_all.rb]: It is ok like that.

File: cbrain_task/recon_all/bourreau/recon_all.rb [hunk 2]
```diff
@@ -58,6 +58,14 @@ def setup #:nodoc:
       self.safe_symlink(cache_path, file.name)
     end
 
+    # Copy personal license for FreeSurfer
+    license = FreesurferLicense.find_all_accessible_by_user(User.find(self.user_id)).first
```
Comment (prioux) [inline cbrain_task/recon_all/bourreau/recon_all.rb]: Some of the code is unnecessary:

* Use the assoc: `find_all_accessible_by_user(self.user)`
* Use the object: `make_available(license, "license.txt")`

File: cbrain_task/recon_all/bourreau/recon_all.rb [hunk 3]
```diff
@@ -139,7 +147,23 @@ def cluster_commands #:nodoc:
 
     recon_all_command = "recon-all#{lbl_ext} #{with_qcache} #{with_mprage} #{with_3T_data} #{with_cw256} #{with_hippocampal} -sd . #{subjid_info} #{step} #{with_notal_check} #{lbl_options}"
 
+    # Copy license
+    cp_license = <<-CP_LICENSE
+
+# Handle FreeSurfer license
+
+if [ ! -f "$FREESURFER_HOME/license.txt" ] && [ ! -f "$FREESURFER_HOME/.license" ] ; then
```
Comment (prioux) [inline cbrain_task/recon_all/bourreau/recon_all.rb]: What is that test for `$FREESRUFER_HOME/.license` ? Is this an alternative way to provide a license? And does it have to be a file or a directory? (The -f will only be true if it's a file)

File: cbrain_task/recon_all/bourreau/recon_all.rb [hunk 4]
```diff
@@ -139,7 +147,23 @@ def cluster_commands #:nodoc:
 
     recon_all_command = "recon-all#{lbl_ext} #{with_qcache} #{with_mprage} #{with_3T_data} #{with_cw256} #{with_hippocampal} -sd . #{subjid_info} #{step} #{with_notal_check} #{lbl_options}"
 
+    # Copy license
+    cp_license = <<-CP_LICENSE
```
Comment (prioux) [inline cbrain_task/recon_all/bourreau/recon_all.rb]: Usually using the `-HERE` syntax is to allow the token to be indented, yet it isn't indented later on.

File: cbrain_task/recon_all/bourreau/recon_all.rb [hunk 5]
```diff
@@ -139,7 +147,23 @@ def cluster_commands #:nodoc:
...
+
+# Handle FreeSurfer license
+
+if [ ! -f "$FREESURFER_HOME/license.txt" ] && [ ! -f "$FREESURFER_HOME/.license" ] ; then
+  echo Could not find a license installed.
+  if test -f license.txt ; then
+    echo Attempting to install the license
+    cp license.txt "$FREESURFER_HOME"
```
Comment (prioux) [inline cbrain_task/recon_all/bourreau/recon_all.rb]: Maybe exit immediately if the cp command fails?

`cp license.txt "$FREESURFER_HOME" || exit 20`

File: cbrain_task/recon_all_longi/bourreau/recon_all_longi.rb [hunk 6]
```diff
@@ -182,7 +206,7 @@ def cluster_commands #:nodoc:
       CAT_SCRIPT
     end
 
-    [ recon_all_base_cmd ] + recon_all_long_cmds + cat_cmds
+    [ cp_license ] + [ recon_all_base_cmd ] + recon_all_long_cmds + cat_cmds
```
Comment (prioux) [inline cbrain_task/recon_all_longi/bourreau/recon_all_longi.rb]: That's a really weird way to build the script....

File: cbrain_task/recon_all/bourreau/recon_all.rb (hunk 3 above)
Comment (natacha-beck) [reply cbrain_task/recon_all/bourreau/recon_all.rb]: It is a file, and yes it is an alternative way to specify a license.

**Pull request merged**